### PR TITLE
Added adf4159 pyadi API with example code

### PR DESCRIPTION
**Pull request by darshiltrivedi**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How has this been tested?
 
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
ADF4159 pyadi API is tested using Eval board EV-ADF4159EB1Z, power supply, spectrum analyzer and DMM. Note that Resistor R9, R18 and R19 on eval board were removed and external 330 Ohms resistor were connected to CLK, MOSI and CS pins of RPI which work as level shifters. Connecting the test setup of adf4159 as mentioned in EVAL-board user guide, the spectrum analyzer shows output frequency that is set in code. Connect a DMM to Vtune and run example code of adf4159 and you can see the Vtune voltage changes every 5 seconds with change in ramp mode. To check enable/powerdown as soon as it is set to 1 spectrum analyzer shows 10.606GHz that is minimum and shows the part is not working.

- [ ] Test B
The adf4159 API was tested locally on Raspberry Pi as well as on a host PC. It worked as expected in both the cases.

**Test Configuration**:
* Hardware:
* OS:

# Documentation

If this is a new feature or example please mention or link any documentation. All new hardware interface classes require documentation.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

@mhennerich @mthoren-adi @tfcollins

### CODE REVIEW

**Comment by tfcollins** [comment]
Can you address the formatting issues. See here for more info: https://analogdevicesinc.github.io/pyadi-iio/dev/index.html#precommit

File: adi/adf4159.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+import numpy as np
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class adf4159(rx, context_manager, attribute):
```
Comment (tfcollins) [inline adi/adf4159.py:41]: PLLs tend not to have buffers or ADC-like data. Is there a specific reason you are using the **rx** class to inherit from?

File: examples/adf4159.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2019 Analog Devices, Inc.
```
Comment (tfcollins) [inline examples/adf4159.py:1]: Update year to 2021

**Comment by tfcollins** [comment]
Can you add a doc page and update supported_parts.md as well

File: adi/adf4159.py (hunk 1 above)
Comment (darshiltrivedi) [reply adi/adf4159.py]: Initially I was trying to make API in a way that we have a room to add properties/attributes

**Comment by tfcollins** [comment]
CI is still showing a formatting issue

**Pull request merged**